sysdeps/managarm/generic/file.cpp

```diff
@@ -1628,12 +1629,14 @@ int sys_mknodat(int dirfd, const char *path, int mode, int dev) {
 }
 
 int sys_read(int fd, void *data, size_t max_size, ssize_t *bytes_read) {
-	SignalGuard sguard;
```

Comment:
Why do we need to do this?

This prevents a signal to be raised, but it should not prevent POSIX from interrupting the thread. I think we still want to keep this.